theme/theme.go

```diff
@@ -61,6 +61,39 @@ func DarkTheme() fyne.Theme {
 	return theme
 }
 
+// shade will darken a light color and lighten a light color by the given % value
+// should use 4,8, 12 or 14 percent darken/lighten to implement google material design rules
+func Shade(c color.Color, pct uint32) color.Color {
+	r, g, b, a := c.RGBA()
+	if pct == 0 {
+		return color.NRGBA{uint8(r), uint8(g), uint8(b), uint8(a)}
+	}
+	if pct > 50 {
+		pct = 50
+	}
+	if r+g+b < 3*0x8080 {
+		// Lighten
+		return color.NRGBA{
+			R: uint8((r + (0x10000-r)*pct/50) >> 8),
+			G: uint8((g + (0x10000-g)*pct/50) >> 8),
+			B: uint8((b + (0x10000-b)*pct/50) >> 8),
+			A: uint8(a >> 8),
+		}
+	} else {
+		// Darken
+		return color.NRGBA{
+			R: uint8((r * (100 - pct) / 100) >> 8),
+			G: uint8((g * (100 - pct) / 100) >> 8),
+			B: uint8((b * (100 - pct) / 100) >> 8),
+			A: uint8(a >> 8),
+		}
+	}
+}
+
+const PressedShade = 14
```

Comment:
Are these percentages? should they be public API?

Reply:
Yes, they are percentages. I belive they should be part of the customizable theme, but I do not see how to do it.